src/pudl/output/eia860.py

```diff
@@ -98,6 +98,11 @@ def plants_eia860(pudl_engine, start_date=None, end_date=None):
         pandas.DataFrame: A DataFrame containing all the fields of the EIA 860
         Plants table.
     """
+    logger.warning(
+        "pudl.output.eia860.plants_eia860() will be deprecated in a future version of PUDL."
+        " In the future, call the PudlTabl.plants_eia860() method or pull the denorm_plants_eia table"
+        "directly from the pudl.sqlite database."
+    )
```

Comment:
We intend to deprecate the `PudlTabl` object entirely, so we shouldn't be pointing people at that solution going forward. It's eventually gonna be a database table, period. I'm not sure if we want to mention the name of the database table though since I think we have yet to make a decision about how the tables are going to be named going forward.  If we expect the `denorm_` tables to be the primary points of access for users in our "Data Warehouse", then probably we'll want to ditch the prefix so the primary point of data access has the simplest most legible name.

If we put these warnings in every function I think they're going to become overwhelming.  Given that we're going to deprecate `PudlTabl` and mostly we read through those objects right now (and so do other people) maybe the right thing to do is put a `warnings.warn()` in `PudlTabl.__init__()` so that it comes up when anyone creates the object.

Why do we need to largely duplicate the contents of this function in `denorm_plants_eia()` rather than converting this function into the asset directly or separating the functionality as needed between the two? This seems dangerous (hard to maintain, likely to make it ambiguous where information is coming from -- not that we want this to stick around for a long time but... famous last words).

If need be we can name the asset whatever we want, independent of the name of the function itself with additional parameters in the `@asset()` decorator. But maybe there's a design constraint I'm not seeing.

Reply:
@bendnorman and I discussed this and decided that we'll try removing the old functions after verifying that the outputs are functionally identical, but it if that ends up being a complicated mess because the intermediate situation with some functions transitioned and others not is too complex, we'll defer removing the old functions until all of the new functions are in place.